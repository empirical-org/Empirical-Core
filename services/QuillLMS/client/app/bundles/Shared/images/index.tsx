export const bigCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-check-big.svg`,
  alt: "Green check icon"
}

export const searchMapIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-map-search.svg`,
  alt: "Map with a magnifying glass icon"
}

export const clipboardCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-clipboard-check.svg`,
  alt: "Clipboard with a checkmark icon"
}

export const tableCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-table-check.svg`,
  alt: "Table with a checkmark icon"
}

export const accountViewIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-view.svg`,
  alt: "Avatar with an eye next to it icon"
}

export const demoViewIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-teacher-account-view.svg`,
  alt: "Teacher preview icon"
}

export const giftIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-gift.svg`,
  alt: "Gift icon"
}

export const groupAccountIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-group.svg`,
  alt: "Three avatars icon"
}

export const googleClassroomIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-google-classroom.svg`,
  alt: "Google Classroom icon"
}

export const helpIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-help.svg`,
  alt: "Question mark icon"
}

export const arrowPointingRightIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-arrow-back.svg`,
  alt: "Arrow pointing right icon"
}

export const expandIcon = {
  src: `${process.env.CDN_URL}/images/icons/expand.svg`,
  alt: 'Expand icon'
}

export const fileChartIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-file-chart.svg`,
  alt: 'File with a chart icon'
}

export const fileDownloadIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-file-download.svg`,
  alt: 'File with a download icon'
}

export const horizontalDotsIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-more-horizontal.svg`,
  alt: 'Three horizontal dots icon'
}

export const playBoxIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-play-box.svg`,
  alt: 'Square play button icon'
}

export const previewIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-preview.svg`,
  alt: 'Eye icon'
}

export const smallWhiteCheckIcon = {
  src: `${process.env.CDN_URL}/images/shared/check-small-white.svg`,
  alt: 'Check icon'
}

export const indeterminateCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/indeterminate.svg`,
  alt: 'Indeterminate check icon'
}

export const cursorClick = {
  src: `${process.env.CDN_URL}/images/icons/cursor-click.svg`,
  alt: 'Default cursor showing exaggerated clicking'
}

export const cursorPointingHand = {
  src: `${process.env.CDN_URL}/images/icons/cursor-pointinghand.svg`,
  alt: 'Cursor pointing hand'
}

export const removeIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-remove.svg`,
  alt: 'Icon showing a gray circle with a horizontal white line inside of it'
}

export const encircledWhiteArrowIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-arrow-right-circle-white.svg`,
  alt: 'Right arrow in a circle'
}

export const greenCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/check-green.svg`,
  alt: 'Green check icon'
}

export const whiteCheckGreenBackgroundIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-check-circle-dark.svg`,
  alt: 'White check with green background icon'
}

export const lockedIcon = {
  alt: 'Lock icon',
  src: `${process.env.CDN_URL}/images/icons/icons-locked.svg`
}

export const closeIcon = {
  alt: 'Close icon',
  src: `${process.env.CDN_URL}/images/icons/close.svg`
}

export const informationIcon = {
  alt: 'Information icon',
  src: `${process.env.CDN_URL}/images/pages/evidence/icons-information-small.svg`
}

export const connectToolIcon = {
  alt: 'Target representing Quill Connect',
  src: `${process.env.CDN_URL}/images/icons/connect-forest-green.svg`
}

export const diagnosticToolIcon = {
  alt: 'Magnifying glass representing Quill Diagnostic',
  src: `${process.env.CDN_URL}/images/icons/diagnostic-forest-green.svg`
}

export const grammarToolIcon = {
  alt: 'Puzzle piece representing Quill Grammar',
  src: `${process.env.CDN_URL}/images/icons/grammar-forest-green.svg`
}

export const lessonsToolIcon = {
  alt: 'Apple representing Quill Lessons',
  src: `${process.env.CDN_URL}/images/icons/lessons-forest-green.svg`
}

export const proofreaderToolIcon = {
  alt: 'Flag representing Quill Proofreader',
  src: `${process.env.CDN_URL}/images/icons/proofreader-forest-green.svg`
}

export const evidenceToolIcon = {
  alt: 'Book representing Quill Reading for Evidence',
  src: `${process.env.CDN_URL}/images/icons/tool-evidence.svg`
}

export const warningIcon = {
  alt: 'Warning icon',
  src: `${process.env.CDN_URL}/images/icons/warning-icon.svg`
}

export const disclosureIcon = {
  alt: '',
  src: `${process.env.CDN_URL}/images/icons/disclosure.svg`
}

export const scheduledIcon = {
  alt: 'Clock icon',
  src: `${process.env.CDN_URL}/images/icons/scheduled.svg`
}

export const publishedIcon = {
  alt: 'Checkmark icon',
  src: `${process.env.CDN_URL}/images/icons/published.svg`
}

export const copyIcon = {
  alt: 'Two identical rectangles, one on top of the other',
  src: `${process.env.CDN_URL}/images/icons/copy_icon.svg`
}

export const assignedBadgeIconGray = {
  alt: 'Gray silhouette of person with checkmark icon',
  src: `${process.env.CDN_URL}/images/icons/assigned_badge.svg`
}

export const assignedBadgeIconWhite = {
  alt: 'White silhouette of person with checkmark icon',
  src: `${process.env.CDN_URL}/images/icons/assigned_badge_white.svg`
}

export const infoIcon = {
  alt: '',
  src: `${process.env.CDN_URL}/images/icons/info.svg`
}

export const closedLockIcon = {
  alt: 'Closed lock',
  src: `${process.env.CDN_URL}/images/icons/locked.svg`
}

export const openLockIcon = {
  alt: 'Open lock',
  src: `${process.env.CDN_URL}/images/icons/unlocked.svg`
}

export const networkIcon = {
  alt: 'Black silhouette of group of 4 people',
  src: `${process.env.CDN_URL}/images/onboarding/network.svg`
}

export const whiteDiamondIcon = {
  alt: 'White diamond',
  src: `${process.env.CDN_URL}/images/icons/white-diamond.svg`
}
export const redDiamondIcon = {
  alt: 'Red diamond',
  src: `${process.env.CDN_URL}/images/icons/red-diamond.svg`
}

export const evidenceHandbookIcon = {
  alt: 'Stack of papers with text "Quill Reading for Evidence Teacher Handbook"',
  src: `${process.env.CDN_URL}/images/evidence/evidence-teacher-handbook.svg`
}
