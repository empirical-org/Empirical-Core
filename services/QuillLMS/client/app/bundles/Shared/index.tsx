export {
  QuestionList,
  LinkListItem,
  QuestionListByConcept
} from './components/questionList/index'

export {
  Instructions,
  Prompt
} from './components/fillInBlank/index'

export {
  ConceptExplanation,
} from './components/feedback/conceptExplanation'

export {
  ReactTable,
  TextFilter,
  expanderColumn,
  NumberFilterInput,
} from './components/reactTable/reactTable'

export {
  ArchivedButton,
  PostNavigationBanner,
  ButtonLoadingSpinner,
  DarkButtonLoadingSpinner,
  LightButtonLoadingSpinner,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  DropdownInputWithSearchTokens,
  Error,
  ExpandableCard,
  FlagDropdown,
  Input,
  OneThumbSlider,
  Passthrough,
  ReportHeader,
  ResumeOrBeginButton,
  ScreenreaderInstructions,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  SortableList,
  TeacherPreviewMenu,
  TeacherPreviewMenuButton,
  TextEditor,
  TextArea,
  ToggleComponentSection,
  Tooltip,
  TwoThumbSlider,
  UploadOptimalResponses
} from './components/shared/index'

export { richButtonsPlugin, } from './components/draftJSRichButtonsPlugin/index'

export {
  Modal
} from './components/modal/modal'

export {
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields
} from './components/questions/index'

export {
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments
} from './components/renderForQuestions/index'

export {
  QuestionRow
} from './components/scoreAnalysis/index'

export {
  MultipleChoice,
  Register,
  PlayTitleCard
} from './components/studentLessons/index'

export {
  TitleCard
} from './components/titleCards/index'

export {
  hashToCollection,
  isValidRegex,
  momentFormatConstants,
  copyToClipboard,
  getLatestAttempt,
  getCurrentQuestion,
  getQuestionsWithAttempts,
  getFilteredQuestions,
  getDisplayedText,
  renderPreviewFeedback,
  roundValuesToSeconds,
  roundMillisecondsToSeconds,
  titleCase,
  onMobile,
  fillInBlankInputLabel,
  getIconForActivityClassification,
  isTrackableStudentEvent,
  hexToRGBA,
  uniqueValuesArray,
  filterNumbers,
  redirectToActivity,
  renderNavList,
  noResultsMessage
} from './libs/index'

export {
  bigCheckIcon,
  searchMapIcon,
  clipboardCheckIcon,
  tableCheckIcon,
  accountViewIcon,
  demoViewIcon,
  giftIcon,
  groupAccountIcon,
  googleClassroomIcon,
  helpIcon,
  arrowPointingRightIcon,
  expandIcon,
  fileChartIcon,
  fileDownloadIcon,
  horizontalDotsIcon,
  playBoxIcon,
  previewIcon,
  smallWhiteCheckIcon,
  indeterminateCheckIcon,
  cursorClick,
  cursorPointingHand,
  removeIcon,
  encircledWhiteArrowIcon,
  greenCheckIcon,
  whiteCheckGreenBackgroundIcon,
  lockedIcon,
  closeIcon,
  informationIcon,
  connectToolIcon,
  diagnosticToolIcon,
  grammarToolIcon,
  lessonsToolIcon,
  proofreaderToolIcon,
  evidenceToolIcon,
  warningIcon,
  disclosureIcon,
  copyIcon,
  scheduledIcon,
  publishedIcon,
  assignedBadgeIconGray,
  assignedBadgeIconWhite,
  infoIcon,
  closedLockIcon,
  openLockIcon,
  networkIcon,
  whiteDiamondIcon,
  redDiamondIcon,
  evidenceHandbookIcon,
  whiteArrowPointingDownIcon,
  filterIcon,
  documentFileIcon,
  singleUserIcon,
  whiteEmailIcon,
} from './images/index'

export {
  KEYDOWN,
  MOUSEMOVE,
  MOUSEDOWN,
  CLICK,
  KEYPRESS,
  VISIBILITYCHANGE,
  SCROLL,
  TOUCHMOVE,
} from './utils/eventNames'

export {
  DEFAULT_HIGHLIGHT_PROMPT,
  BECAUSE,
  BUT,
  SO,
  READ_PASSAGE_STEP_NUMBER,
  BECAUSE_PASSAGE_STEP_NUMBER,
  BUT_PASSAGE_STEP_NUMBER,
  SO_PASSAGE_STEP_NUMBER,
  EVIDENCE,
  CONNECT,
  DIAGNOSTIC,
  GRAMMAR,
  LESSONS,
  PROOFREADER,
  CURRICULUM,
  PARTNERSHIPS,
  PRODUCT,
  SUPPORT,
  TEAMS,
  ARCHIVED_FLAG,
  PRODUCTION_FLAG,
  ALPHA_FLAG,
  BETA_FLAG,
  GAMMA_FLAG,
  PRIVATE_FLAG,
  NOT_APPLICABLE,
  STUDENT,
  TEACHER,
  INDIVIDUAL_CONTRIBUTOR,
  ADMIN,
  NOT_LISTED,
  NO_SCHOOL_SELECTED,
  EVIDENCE_HANDBOOK_LINK,
  MAX_VIEW_WIDTH_FOR_MOBILE_NAVBAR,
  INTRODUCTION,
  CHECKLIST,
  READ_AND_HIGHLIGHT
} from './utils/constants'

export { DefaultReactQueryClient } from './utils/defaultReactQueryClient'
