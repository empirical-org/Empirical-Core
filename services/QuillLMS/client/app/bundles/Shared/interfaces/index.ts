export { UserIdsForEvent, NumberFilterInputProps } from './misc'
export { Question, QuestionObject } from './question'
