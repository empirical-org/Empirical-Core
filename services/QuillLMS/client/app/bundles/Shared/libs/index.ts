export { hashToCollection } from './hashToCollection'
export { isValidRegex } from './isValidRegex'
export { momentFormatConstants } from './momentFormatConstants'
export { copyToClipboard } from './copyToClipboard'
export { getLatestAttempt } from './getLatestAttempt'
export { getCurrentQuestion, getQuestionsWithAttempts, getFilteredQuestions, getDisplayedText, renderPreviewFeedback } from './activityPreviewHelpers'
export { roundValuesToSeconds, roundMillisecondsToSeconds, } from './timeTrackingSharedFunctions'
export { titleCase } from './titleCase'
export { onMobile } from './onMobile'
export { fillInBlankInputLabel, } from './fillInBlankInputLabel'
export { getIconForActivityClassification } from './getIconForActivityClassification'
export { commaSeparatedValuesString, } from './commaSeparatedValuesString'
