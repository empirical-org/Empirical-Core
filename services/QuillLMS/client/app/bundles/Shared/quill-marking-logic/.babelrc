{
  "presets": ["es2015"],
  "plugins": ["transform-object-rest-spread", "syntax-dynamic-import"],
  "env": {
    "test": {
      "plugins": ["transform-es2015-modules-commonjs", "dynamic-import-node"]
    }
  }
}
