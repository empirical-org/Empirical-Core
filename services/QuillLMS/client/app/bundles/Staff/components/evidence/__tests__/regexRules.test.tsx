import * as React from 'react';
import { shallow } from 'enzyme';

import { Input } from '../../../../Shared/index';
import RegexRules from '../configureRules/regexRules';
jest.mock('string-strip-html', () => ({
  default: jest.fn()
}))

const mockRegexRules = {
  'regex-rule-0': {
    id: 1,
    regex_text: 'it contain(s)? methane gas',
    case_sensitive: false
  },
  'regex-rule-1': {
    id: 2,
    regex_text: 'another reg(ex) line',
    case_sensitive: true
  },
  'regex-rule-2': {
    id: 3,
    regex_text: 'some m?ore reg(ex',
    case_sensitive: false
  }
}
const mockProps = {
  errors: {},
  handleAddRegexInput: jest.fn(),
  handleDeleteRegexRule: jest.fn(),
  handleSetRegexRule: jest.fn(),
  handleSetRegexRuleSequence: jest.fn(),
  regexRules: mockRegexRules
};

describe('RegexRules component', () => {
  const container = shallow(<RegexRules {...mockProps} />);

  it('should render RegexRules', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render an Input for each field', () => {
    // Input: Regex Rules (3)
    // input of type checkbox (3)
    expect(container.find(Input).length).toEqual(3);
  });
  it('should render a checkbox for each regex rule that has case sensitivity', () => {
    expect(container.find('#regex-case-sensitive-0').props().checked).toEqual(false);
    expect(container.find('#regex-case-sensitive-1').props().checked).toEqual(true);
    expect(container.find('#regex-case-sensitive-2').props().checked).toEqual(false);
  });
  it('clicking each checkbox should call handleSetRegexRule', () => {
    container.find('#regex-case-sensitive-0').simulate('change', { target: { checked: true }});
    container.find('#regex-case-sensitive-1').simulate('change', { target: { checked: true }});
    container.find('#regex-case-sensitive-2').simulate('change', { target: { checked: true }});
    expect(mockProps.handleSetRegexRule).toHaveBeenCalledTimes(3);
  });
  it('clicking each delete button should call handleDeleteRegexRule', () => {
    container.find('#remove-regex-button').at(0).simulate('click');
    container.find('#remove-regex-button').at(1).simulate('click');
    container.find('#remove-regex-button').at(2).simulate('click');
    expect(mockProps.handleDeleteRegexRule).toHaveBeenCalledTimes(3);
  });
  it('clicking add regex button should call handleAddRegexInput', () => {
    container.find('#add-regex-button').simulate('click');
    expect(mockProps.handleAddRegexInput).toHaveBeenCalled();
  });
});
