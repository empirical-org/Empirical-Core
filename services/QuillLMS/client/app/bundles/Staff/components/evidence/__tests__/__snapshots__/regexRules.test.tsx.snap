// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegexRules component should render RegexRules 1`] = `
<div
  className="regex-rules-container"
>
  <div
    className="regex-rule-container"
    key="regex-rule-container-0"
  >
    <div
      className="regex-input-container"
    >
      <Input
        autoComplete="on"
        className="regex-input"
        handleChange={[MockFunction]}
        id="regex-rule-0"
        value="it contain(s)? methane gas"
      />
      <p
        className="characters-remaining"
      >
        174
         characters remaining
      </p>
      <div
        className="checkbox-container"
      >
        <DropdownInput
          className="rule-type-input"
          handleChange={[Function]}
          id="regex-rule-0"
          isSearchable={true}
          label="Sequence Type"
          options={
            Array [
              Object {
                "label": "Incorrect",
                "value": "incorrect",
              },
              Object {
                "label": "Required",
                "value": "required",
              },
            ]
          }
          value={
            Object {
              "label": "Incorrect",
              "value": "incorrect",
            }
          }
        />
        <label
          className="case-sensitive-label"
          htmlFor="regex-case-sensitive-0"
        >
          Case Sensitive?
        </label>
        <input
          aria-labelledby="regex-rule-0"
          checked={false}
          id="regex-case-sensitive-0"
          onChange={[MockFunction]}
          type="checkbox"
          value="regex-rule-0"
        />
      </div>
    </div>
    <button
      className="quill-button fun primary outlined delete-regex-button"
      id="remove-regex-button"
      onClick={[MockFunction]}
      type="button"
      value="regex-rule-0"
    >
      remove
    </button>
  </div>
  <div
    className="regex-rule-container"
    key="regex-rule-container-1"
  >
    <div
      className="regex-input-container"
    >
      <Input
        autoComplete="on"
        className="regex-input"
        handleChange={[MockFunction]}
        id="regex-rule-1"
        value="another reg(ex) line"
      />
      <p
        className="characters-remaining"
      >
        180
         characters remaining
      </p>
      <div
        className="checkbox-container"
      >
        <DropdownInput
          className="rule-type-input"
          handleChange={[Function]}
          id="regex-rule-1"
          isSearchable={true}
          label="Sequence Type"
          options={
            Array [
              Object {
                "label": "Incorrect",
                "value": "incorrect",
              },
              Object {
                "label": "Required",
                "value": "required",
              },
            ]
          }
          value={
            Object {
              "label": "Incorrect",
              "value": "incorrect",
            }
          }
        />
        <label
          className="case-sensitive-label"
          htmlFor="regex-case-sensitive-1"
        >
          Case Sensitive?
        </label>
        <input
          aria-labelledby="regex-rule-1"
          checked={true}
          id="regex-case-sensitive-1"
          onChange={[MockFunction]}
          type="checkbox"
          value="regex-rule-1"
        />
      </div>
    </div>
    <button
      className="quill-button fun primary outlined delete-regex-button"
      id="remove-regex-button"
      onClick={[MockFunction]}
      type="button"
      value="regex-rule-1"
    >
      remove
    </button>
  </div>
  <div
    className="regex-rule-container"
    key="regex-rule-container-2"
  >
    <div
      className="regex-input-container"
    >
      <Input
        autoComplete="on"
        className="regex-input"
        handleChange={[MockFunction]}
        id="regex-rule-2"
        value="some m?ore reg(ex"
      />
      <p
        className="characters-remaining"
      >
        183
         characters remaining
      </p>
      <div
        className="checkbox-container"
      >
        <DropdownInput
          className="rule-type-input"
          handleChange={[Function]}
          id="regex-rule-2"
          isSearchable={true}
          label="Sequence Type"
          options={
            Array [
              Object {
                "label": "Incorrect",
                "value": "incorrect",
              },
              Object {
                "label": "Required",
                "value": "required",
              },
            ]
          }
          value={
            Object {
              "label": "Incorrect",
              "value": "incorrect",
            }
          }
        />
        <label
          className="case-sensitive-label"
          htmlFor="regex-case-sensitive-2"
        >
          Case Sensitive?
        </label>
        <input
          aria-labelledby="regex-rule-2"
          checked={false}
          id="regex-case-sensitive-2"
          onChange={[MockFunction]}
          type="checkbox"
          value="regex-rule-2"
        />
      </div>
    </div>
    <button
      className="quill-button fun primary outlined delete-regex-button"
      id="remove-regex-button"
      onClick={[MockFunction]}
      type="button"
      value="regex-rule-2"
    >
      remove
    </button>
  </div>
  <button
    className="quill-button fun primary outlined add-regex-button"
    id="add-regex-button"
    onClick={[MockFunction]}
    type="button"
  >
    Add Regex Rule +
  </button>
</div>
`;
